tests/Doctrine/Tests/ORM/Functional/Ticket/GH10348Test.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\Common\Collections\Collection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+final class GH10348Test extends OrmFunctionalTestCase
+{
+    public function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH10348Person::class,
+            GH10348Company::class,
+        ]);
+    }
+
+    public function testCanRemoveParentWithSelfReferencingChild(): void
```

Comment:
Maybe by changing the title of the test again?
```suggestion
    public function testTheORMRemovesReferencedEmployeeBeforeReferencingEmployee(): void
```